src/SymfonyCache/TaggableStore.php

```diff
@@ -0,0 +1,387 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCache\SymfonyCache;
+
+use Psr\Cache\InvalidArgumentException;
+use Symfony\Component\Cache\Adapter\FilesystemAdapter;
+use Symfony\Component\Cache\Adapter\TagAwareAdapter;
+use Symfony\Component\HttpFoundation\HeaderBag;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\HttpCache\StoreInterface;
+use Symfony\Component\Lock\Exception\LockReleasingException;
+use Symfony\Component\Lock\Factory;
+use Symfony\Component\Lock\LockInterface;
+use Symfony\Component\Lock\Store\FlockStore;
+
+/**
+ * Implements a storage for Symfony's HttpCache that supports tagging.
+ *
+ * @author Yanick Witschi <yanick.witschi@terminal42.ch>
+ */
+class TaggableStore implements StoreInterface
+{
+    const NON_VARYING_KEY = 'non-varying';
+
+    /**
+     * @var string
+     */
+    private $purgeTagsHeader;
+
+    /**
+     * @var TagAwareAdapter
+     */
+    private $cache;
+
+    /**
+     * @var Factory
+     */
+    private $lockFactory;
+
+    /**
+     * @var LockInterface[]
+     */
+    private $locks = [];
+
+    /**
+     * @param string $cacheDir
+     */
+    public function __construct($cacheDir, $purgeTagsHeader = PurgeListener::DEFAULT_PURGE_TAGS_HEADER)
+    {
+        if (!class_exists(Factory::class)) {
+            throw new \RuntimeException('You need at least Symfony 3.4 to use the TaggableStore.');
+        }
+
+        $this->purgeTagsHeader = $purgeTagsHeader;
+        $this->cache = $cache = new TagAwareAdapter(new FilesystemAdapter('fos-http-cache', 0, $cacheDir));
+        $this->lockFactory = new Factory(new FlockStore($cacheDir));
+    }
+
+    /**
+     * @return TagAwareAdapter
+     */
+    public function getCache()
```

Comment:
Basically this is only needed for unit tests but it doesn't harm either.

Reply:
i would use reflection in the unit test rather than expose this. we don't want people to interact with the tag cache from outside.

Reply:
I agree. Fixed in d56f6a43b573eb72bd7733bdb02785547a6e5df1.